tools/deploymentsSchemaTests/TestSchemaCache.cs

```diff
@@ -0,0 +1,166 @@
+﻿using Azure.Deployments.Core.Components;
+using Azure.Deployments.Core.Extensions;
+using Azure.Deployments.Core.Resources;
+using Azure.Deployments.Templates.Contracts;
+using Azure.Deployments.Templates.Export;
+using Azure.Deployments.Templates.Extensions;
+using Azure.Deployments.Templates.Helpers;
+using Azure.Deployments.TemplateSchemas;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace deploymentsSchemaTests
+{
+    /// <summary>
+    /// Test schema cache
+    /// </summary>
+    internal class TestSchemaCache : INormalizedSchemaCache
```

Comment:
>TestSchemaCache [](start = 19, length = 15)

Had to create a custom class, for the most part this code is identical to the OfflineSchemaCache. When I tried to use CreateForNormalizedSchemas (and passed all schemas folders) I got an error saying that some schemas (37 actually) were not found, some of them had to do with CreateUIDefinition.CommonControl.json (the ones from 0.0.1-preview and 0.X.X-preview), so my approach was to use the same code CreateFromAssembly implements with the difference that instead of loading schemas from the assembly I use filePaths instead.